EspressoSystems/HotShot
# [TECH DEBT] Refactor errors

### This PR: 






Is the first step towards getting better logging/errors.

- Switches everything from `snafu` to `thiserror`. This should get rid of the errors akin to "Libp2p-specific errors" and "Error in the Kademlia DHT"
- Deduplicates some warnings/errors. Some errors will log and then return, where they got logged a second time. I plan to do more of this in a coming PR, along with adding more context where necessary
- Removes unused errors and error variants


### This PR does not: 


- Refine log levels
- Change things to use `anyhow` anywhere
- Apart from formatting, does not touch existing, in-use log messages (besides adding context) or log types. 


### Key places to review: 


The only really important errors are the HotShot ones, so everything mentioning `HotShotError`

### How to test this PR: 




I've run this in the sequencer. I didn't see any unused error variants or anything
